File: thisboot/src/main/java/com/brageast/blog/thisboot/service/ArticleService.java
package com.brageast.blog.thisboot.service;

import com.brageast.blog.thisboot.entity.Article;
import com.mongodb.client.result.DeleteResult;
import com.mongodb.client.result.UpdateResult;
import org.bson.types.ObjectId;
import reactor.core.publisher.Flux;
import reactor.core.publisher.Mono;

public interface ArticleService {
    Flux<Article> limitShow(int page, int size);

    Mono<Article> insert(Mono<Article> article);

    Mono<Article> insert(Article article);

    Flux<Article> show();

    Mono<DeleteResult> delete(ObjectId id);

    Mono<UpdateResult> update(Article article);

    Mono<Article> findById(ObjectId articleId);
}

File: thisboot/src/main/kotlin/com/brageast/blog/thisboot/service/ArticleService.kt
package com.brageast.blog.thisboot.service

import com.brageast.blog.thisboot.entity.Article
import org.bson.types.ObjectId
import org.springframework.stereotype.Service
import reactor.core.publisher.Flux
import reactor.core.publisher.Mono

interface ArticleService {
    fun insert(article: Article): Mono<Article>
    fun findAll(): Flux<Article>
    fun findWithPagination(page: Int, size: Int): Flux<Article>
    fun updateOrInsert(article: Article): Mono<Article>
    fun findById(id: String): Mono<Article>
    fun findById(id: ObjectId): Mono<Article>
    fun deleteById(id: ObjectId): Mono<Void>
    fun getSize(): Mono<Long>
}
